### PR TITLE
[Issue #390]added the client._generate._get_known_symbols

### PR DESCRIPTION
**Pull request by muddi900**
Fixes: Issue #390, uses pre-PEP585 typing annotations.

edit: gitignores for local ide settings and test outputs

### CODE REVIEW

File: comtypes/client/_generate.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 from comtypes.tools import codegenerator, tlbparser
 
 if TYPE_CHECKING:
-    from typing import Any, Tuple, List, Optional, Union as _UnionT
+    from typing import Any, Tuple, List, Optional, Dict, Union as _UnionT
```
Comment (junkmd) [inline comtypes/client/_generate.py:20]: 👍

File: comtypes/client/_generate.py [hunk 2]
```diff
@@ -113,17 +113,19 @@ def GetModule(tlib):
         # the directory of the calling module (if not from command line)
         frame = sys._getframe(1)
         _file_ = frame.f_globals.get("__file__", None)  # type: str
-        pathname, is_abs = _resolve_filename(tlib_string, _file_ and os.path.dirname(_file_))
+        pathname, is_abs = _resolve_filename(
+            tlib_string, _file_ and os.path.dirname(_file_))
```
Comment (junkmd) [inline comtypes/client/_generate.py:117]: Please revert it since it is not related to #390

File: comtypes/client/_generate.py [hunk 3]
```diff
@@ -113,17 +113,19 @@ def GetModule(tlib):
...
         logger.debug("GetModule(%s), resolved: %s", pathname, is_abs)
         tlib = _load_tlib(pathname)  # don't register
         if not is_abs:
             # try to get path after loading, but this only works if already registered
             pathname = tlbparser.get_tlib_filename(tlib)
             if pathname is None:
-                logger.info("GetModule(%s): could not resolve to a filename", tlib)
+                logger.info(
+                    "GetModule(%s): could not resolve to a filename", tlib)
```
Comment (junkmd) [inline comtypes/client/_generate.py:125]: Please revert it since it is not related to #390

File: comtypes/client/_generate.py [hunk 4]
```diff
@@ -113,17 +113,19 @@ def GetModule(tlib):
...
                 pathname = tlib_string
         # if above path torture resulted in an absolute path, then the file exists (at this point)!
-        assert not(os.path.isabs(pathname)) or os.path.exists(pathname)
+        assert not (os.path.isabs(pathname)) or os.path.exists(pathname)
```
Comment (junkmd) [inline comtypes/client/_generate.py:128]: Please revert it since it is not related to #390

File: comtypes/client/_generate.py [hunk 5]
```diff
@@ -162,7 +164,8 @@ def _load_tlib(obj):
         libid, ver = obj[0], obj[1:]
         if not ver:  # case of version numbers are not containing
             with winreg.OpenKey(winreg.HKEY_CLASSES_ROOT, r"TypeLib\%s" % libid) as key:
-                ver = [int(v, base=16) for v in winreg.EnumKey(key, 0).split(".")]
+                ver = [int(v, base=16)
+                       for v in winreg.EnumKey(key, 0).split(".")]
```
Comment (junkmd) [inline comtypes/client/_generate.py:168]: Please revert it since it is not related to #390

File: comtypes/client/_generate.py [hunk 6]
```diff
@@ -246,9 +249,9 @@ def _create_wrapper_module(tlib, pathname):
     return _create_module_in_file(modulename, code)
 
 
-def _get_known_symbols():
-    # type: () -> dict[str, str]
-    known_symbols = {}  # type: dict[str, str]
+def _get_known_symbols() -> Dict[str, str]:
+
+    known_symbols: Dict[str, str] = {}
```
Comment (junkmd) [inline comtypes/client/_generate.py:254]: This package still supports Python 2.7.

You should write type annotations by comments in the way of [PEP484](https://peps.python.org/pep-0484/#suggested-syntax-for-python-2-7-and-straddling-code).

For example, here will be...
```py
def _get_known_symbols():
    # type: () -> Dict[str, str]
    known_symbols = {}  # type: Dict[str, str]
    ...
```

File: comtypes/client/_generate.py [hunk 7]
```diff
@@ -259,7 +262,7 @@ def _get_known_symbols():
     ):
         mod = importlib.import_module(mod_name)
         if hasattr(mod, "__known_symbols__"):
-            names = mod.__known_symbols__  # type: list[str]
+            names: List[str] = mod.__known_symbols__
```
Comment (junkmd) [inline comtypes/client/_generate.py:265]: You should write type annotations by comments in the way of [PEP484](https://peps.python.org/pep-0484/#suggested-syntax-for-python-2-7-and-straddling-code).

**Review by junkmd — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 8]
```diff
@@ -51,3 +51,7 @@ coverage.xml
 docs/_build/
 
 
+custom location
```
Comment (junkmd) [inline .gitignore:54]: Please revert here.

**Review by junkmd — CHANGES_REQUESTED** [review]

File: comtypes/client/_generate.py [hunk 9]
```diff
@@ -247,8 +247,9 @@ def _create_wrapper_module(tlib, pathname):
 
 
 def _get_known_symbols():
-    # type: () -> dict[str, str]
-    known_symbols = {}  # type: dict[str, str]
+    # type: () -> Dict[str,str]
+
```
Comment (junkmd) [inline comtypes/client/_generate.py:251]: Please delete this blank line.

File: comtypes/client/_generate.py [hunk 10]
```diff
@@ -247,8 +247,9 @@ def _create_wrapper_module(tlib, pathname):
...
+    known_symbols = {} # type: Dict[str, str]
```
Comment (junkmd) [inline comtypes/client/_generate.py:252]: Please use two spaces before a comments

File: comtypes/client/_generate.py [hunk 11]
```diff
@@ -259,7 +260,7 @@ def _get_known_symbols():
     ):
         mod = importlib.import_module(mod_name)
         if hasattr(mod, "__known_symbols__"):
-            names = mod.__known_symbols__  # type: list[str]
+            names = mod.__known_symbols__ # type: List[str]
```
Comment (junkmd) [inline comtypes/client/_generate.py:263]: Please use two spaces before a comments

**Review by junkmd — CHANGES_REQUESTED** [review]

**Review by junkmd — CHANGES_REQUESTED** [review]
Your environment only setting are should not add to the PR.

File: comtypes/client/_generate.py [hunk 12]
```diff
@@ -247,8 +247,8 @@ def _create_wrapper_module(tlib, pathname):
 
 
 def _get_known_symbols():
-    # type: () -> dict[str, str]
-    known_symbols = {}  # type: dict[str, str]
+     # type: () -> Dict[str,str]
```
Comment (junkmd) [inline comtypes/client/_generate.py:250]: Strange spacing...
Please fix it.
```py
def _get_known_symbols():
    # type: () -> Dict[str, str]
    ...
```

**Review by junkmd — CHANGES_REQUESTED** [review]
And please do such as `git rm "custom location" -r`.

"Your environment only setting are should not add to the PR." is **no** means to say that "You can add your non-ignored test outputs to the PR freely".

**Comment by junkmd** [comment]
@muddi900 

If you are not sure what the commands mean or what the final destination should be, please feel free to ask questions.

**Comment by muddi900** [comment]
> @muddi900
> 
> If you are not sure what the commands mean or what the final destination should be, please feel free to ask questions.

Is there a contribution doc? I couldn't find it in the docs.

**Comment by junkmd** [comment]
@muddi900 

> a contribution doc?

I'm working on that, please see #387.

I have recently been attempting to increase contributors in this community.

Therefore, there are still imperfections. Sorry.

**Comment by junkmd** [comment]
@muddi900 

As a recommendation, you should deactivate the auto-formatter in VSCode's workspace settings.

Applying your formatter to a project that does not use a formatter or uses a different type of formatter and settings, such as this project, can cause unexpected changes.

**Review by junkmd — APPROVED** [review]
Type annotations are fixed. Thanks @muddi900 

<details>
<summary>misc</summary> 

https://ci.appveyor.com/project/pywinauto/pywinauto/builds/45610244
</details>

**Pull request merged**